src/app/services/actions/permissions-action-creator.ts

```diff
@@ -207,6 +209,18 @@ export function consentToScopes(scopes: string[]) {
   };
 }
 
+const correctConsentedScopes = (scopeToBeConsented: string[], consentedScopes: string[],
```

Comment:
Naming: the function name is unclear. Does it correct the consented scopes or return a set of consented scopes?

Reply:
It validates the consented scopes. How about 
const validatedScopes = validateConsentedScopes()
The function checks whether all previously consented scopes are available.